### PR TITLE
Support filedisk.Disk instances as well as image paths

### PR DESCRIPTION
**Pull request by zvin**

### CODE REVIEW

File: build/imagefs.js [hunk 1]
```diff
@@ -82,8 +106,28 @@ exports.read = function(definition) {
...
+      disk = new filedisk.FileDisk(fd, false, false);
+      return write(disk, definition.partition, definition.path, stream).tap(function(writeStream) {
+        writeStream.on('close', close);
+        return writeStream.on('error', close);
+      });
+    });
+  } else if (definition.image instanceof filedisk.Disk) {
+    return write(definition.image, definition.partition, definition.path, stream);
+  } else {
+    throw new Error('image must be a String or a filedisk.Disk instance');
```
Comment (lurch) [inline build/imagefs.js]: I wonder if `... String (filepath) ...` might make this error clearer?

File: lib/imagefs.coffee [hunk 2]
```diff
@@ -72,17 +88,32 @@ exports.read = (definition) ->
...
+			close = -> fs.closeAsync(fd)
+			disk = new filedisk.FileDisk(fd, false, false)
+			write(disk, definition.partition, definition.path, stream)
+			.tap (writeStream) ->
+				writeStream.on('close', close)
+				writeStream.on('error', close)
+	else if definition.image instanceof filedisk.Disk
+		write(definition.image, definition.partition, definition.path, stream)
+	else
+		throw new Error('image must be a String or a filedisk.Disk instance')
```
Comment (lurch) [inline lib/imagefs.coffee]: Seems like this error message appears in multiple places - perhaps a shared error-handling function / wrapper could be extracted?

File: build/imagefs.js (hunk 1 above)
Comment (zvin) [inline build/imagefs.js]: right

File: lib/imagefs.coffee [hunk 3]
```diff
@@ -46,17 +52,30 @@ driver = require('./driver')
...
+			close = -> fs.closeAsync(fd)
+			disk = new filedisk.FileDisk(fd, true)
+			read(disk, definition.partition, definition.path)
+			.tap (stream) ->
+				stream.on('end', close)
+				stream.on('error', close)
+	else if definition.image instanceof filedisk.Disk
+		read(definition.image, definition.partition, definition.path)
 
+read = (disk, partition, path) ->
```
Comment (jviotti) [inline lib/imagefs.coffee]: Can we move this function declaration above the function that first uses it?

Comment (zvin) [inline lib/imagefs.coffee]: Done, I also moved all other helpers definitions above the functions that use them in this file.

**Pull request merged**